subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/caching/DefaultCacheLockingManagerIntegrationTest.groovy

```diff
@@ -62,9 +66,10 @@ class DefaultCacheLockingManagerIntegrationTest extends AbstractHttpDependencyRe
         resources[0].assertExists()
         files[0].assertExists()
         files[1].assertExists()
+        metadata[0].assertExists()
     }
 
-    def "cleans up resources and files that were not recently used from caches"() {
+    def "cleans up resources, files and metadata that were not recently used from caches"() {
```

Comment:
Given this, I wonder if we even need to clean up the metadata directories themselves, since they'll be essentially empty. Or is there more in there?

Reply:
The metadata descriptors directory contains a binary descriptor which is about 4 KB for each artifact which isn't a lot. So in sum it will usually be a few MB. Do you think we should keep them?